File: tests/machine/x/clj/in_operator_extended.clj
(ns main)

(defn _equal [a b]
  (cond
    (and (sequential? a) (sequential? b))
      (and (= (count a) (count b)) (every? true? (map _equal a b)))
    (and (map? a) (map? b))
      (and (= (count a) (count b))
           (every? (fn [k] (_equal (get a k) (get b k))) (keys a)))
    (and (number? a) (number? b))
      (= (double a) (double b))
    :else
      (= a b)))

(defn _in [item col]
  (cond
    (and (string? col) (string? item)) (clojure.string/includes? col item)
    (map? col) (contains? col item)
    (sequential? col) (some #(= item %) col)
    :else false))
(declare xs ys m s)

(defn -main []
  (def xs [1 2 3]) ;; list of int
  (def ys (vec (->> (for [x xs :when (_equal (mod x 2) 1)] x)))) ;; list of int
  (println (some #(= 1 %) ys))
  (println (some #(= 2 %) ys))
  (def m {:a 1}) ;; 
  (println (_in "a" m))
  (println (_in "b" m))
  (def s "hello") ;; string
  (println (clojure.string/includes? s "ell"))
  (println (clojure.string/includes? s "foo"))
)

(-main)

File: tests/machine/x/fs/in_operator_extended.fs
open System

type Anon1 = {
    a: int
}
let xs = [1; 2; 3]
let ys = [ for x in xs do if x % 2 = 1 then yield x ]
printfn "%b" (List.contains 1 ys)
printfn "%b" (List.contains 2 ys)
let m = { a = 1 }
printfn "%s" m.ContainsKey "a"
printfn "%s" m.ContainsKey "b"
let s: string = "hello"
printfn "%s" List.contains "ell" s
printfn "%s" List.contains "foo" s
